numpy/core/src/multiarray/dtype_transfer.c

```diff
@@ -488,7 +457,8 @@ wrap_aligned_contig_transfer_function(
             PyArray_StridedUnaryOp *wrapped, NpyAuxData *wrappeddata,
             int init_dest,
             PyArray_StridedUnaryOp **out_stransfer,
-            NpyAuxData **out_transferdata)
+            NpyAuxData **out_transferdata,
+            int out_needs_api)
```

Comment:
Let's put the out arguments last, and insert this new one right after `init_dest`.